lightning/src/chain/keysinterface.rs

```diff
@@ -47,9 +101,9 @@
 /// that txid/index, and any keys or other information required to sign.
 #[derive(Clone, Debug, PartialEq)]
 pub enum SpendableOutputDescriptor {
-	/// An output to a script which was provided via KeysInterface, thus you should already know
-	/// how to spend it. No keys are provided as rust-lightning was never given any keys - only the
-	/// script_pubkey as it appears in the output.
+	/// An output to a script which was provided via KeysInterface directly, either from
+	/// `get_destination_script()` or `get_shutdown_pubkey()`, thus you should already know how to
+	/// spend it. No secret keys are provided as rust-lightning was never given any key.
 	/// These may include outputs from a transaction punishing our counterparty or claiming an HTLC
 	/// on-chain using the payment preimage or after it has timed out.
 	StaticOutput {
```

Comment:
Note this is really in #786. It would feel awkward to tie the scripts to a channel without moving the functions to get them to the `ChannelSigner`? I dont really want to do that, though, as the current setup is great in that the `KeysInterface` implementation is trivial and gives you direct control over all the entropy used and keys used directly (that aren't fixed by the lightning protocol), and the `ChannelSigner` implementation is all about the lightning protocol details.

I guess we could maybe pass the channel information to the functions in `KeysInterface` to get the fields as a middle ground, but ultimately a user probably would just want some kind of integer id to figure out what HD derivation path they used, not channel information, and I'm not really sure how to capture that easily.

Reply:
Oh, this wasn't about the scripts at all.  It's about this struct (`StaticOutput`), which is constructed in `ChannelMonitor` when sweeping a payment to us.  I'm just saying it would be good for the `KeysManager` to know which channel is being swept for housekeeping / policy controls.  `ChannelMonitor` has the derivation ID handy, so it's easy to populate such a field in this struct.

Reply:
Right, it just felt awkward that we pass an id referencing the derivation id when the script/output itself isn't in any way related to a given derivation, the `KeysManager` returns them directly. Best would be to pass in the derivation info to the keysmanager when we get those scripts, I think, but we don't have it yet at that point, afair.

Reply:
OK, there's no rush to get into this in this PR.